config_src/drivers/FMS_cap/MOM_surface_forcing_gfdl.F90

```diff
@@ -446,6 +447,12 @@ subroutine convert_IOB_to_fluxes(IOB, fluxes, index_bounds, Time, valid_time, G,
         call check_mask_val_consistency(IOB%fprec(i-i0,j-j0), G%mask2dT(i,j), i, j, 'fprec', G)
     endif
 
+    if (associated(IOB%seaice_melt)) then
```

Comment:
@marshallward Is this sufficient to make sure that fluxes%seaice_melt is not filled with nonsense if IOB%seaice_melt is not allocated or used by the coupler?

Reply:
I wouldn't say prevent, it only ensures that the potential nonsense from `IOB%seaice_melt` is transferred to `fluxes%seaice_melt`.

The only solution I can see is for `net_FW` to have some awareness of the state of `IOB%seaice_melt`.  If the coupler fills `IOB%seaice_melt`, then it can be used to compute `fluxes%seaice_metl`.  Otherwise, it cannot use `IOB%seaice_melt` (via `fluxes%seaice_melt`).

Zero-initialization might get you out of this, although generally we try to avoid that solution where possible.

Reply:
I am guessing there is no way to check if `IOB%seaice_melt` has been allocated with actual values?

What if we add `use_seaice_melt` to `IOB` as a logical, default it to false and then set it to true in the coupler where `IOB%seaice_melt` was allocated?  Then use that here to see if `seaice_melt` should be used or not in this routine.

Reply:
After some in preson discussion @marshallward confirmed the if associated should be enough to make sure the pointer was allocated in the coupler. There should be no need to change anything to get this working with older versions of the coupler.